lib/nokogiri/xml/node.rb

```diff
@@ -192,16 +198,47 @@ def decorate!
 
       # :section: Manipulating Document Structure
 
-      ###
-      # Add +node_or_tags+ as a child of this Node.
-      #
-      # +node_or_tags+ can be a Nokogiri::XML::Node, a ::DocumentFragment, a ::NodeSet, or a String
-      # containing markup.
-      #
-      # Returns the reparented node (if +node_or_tags+ is a Node), or NodeSet (if +node_or_tags+ is
-      # a DocumentFragment, NodeSet, or String).
-      #
-      # Also see related method +<<+.
+      # :call-seq:
+      #   add_child(object) -> object
+      #
+      # Appends one or more children (as specified by the given +object+) to +self+;
+      # returns +object+.
+      #
+      # When +object+ is an XML::Node of any kind,
+      # adds the node as the last child of +self+;
+      # sets the parent of that node as +self+:
+      #
+      #   doc = XML::Document.new
```

Comment:
Creating a bare Document this way, and calling Node methods on it, feels very foreign to me and I'm not sure it clearly illustrates that we're talking about Nodes and not Documents in this documentation. Especially since Document#add_child has limitations (a document can't have multiple root nodes).

It feels much more natural to me to do something like this instead:

```ruby
doc = XML::Document.parse("<root></root>")
```

which has implications for documentation below ...

Reply:
Au naturelle.